### PR TITLE
Java Runtime: Add cancellation

### PR DESCRIPTION
**Pull request by martinbonnin**

### CODE REVIEW

File: libraries/apollo-runtime-java/src/main/java/com/apollographql/apollo3/runtime/java/interceptor/ApolloDisposable.java [hunk 1]
```diff
@@ -4,4 +4,12 @@ public interface ApolloDisposable {
   void dispose();
 
   boolean isDisposed();
+
+  void addListener(Listener listener);
+
+  void removeCancellationListener(Listener listener);
+
+  interface Listener {
+    void onCancelled();
```
Comment (BoD) [inline libraries/apollo-runtime-java/src/main/java/com/apollographql/apollo3/runtime/java/interceptor/ApolloDisposable.java:13]: Maybe `onDisposed` (or `onDispose`) would make more sense?

**Review by BoD — APPROVED** [review]
👍

File: libraries/apollo-runtime-java/src/main/java/com/apollographql/apollo3/runtime/java/interceptor/ApolloDisposable.java (hunk 1 above)
Comment (martinbonnin) [reply libraries/apollo-runtime-java/src/main/java/com/apollographql/apollo3/runtime/java/interceptor/ApolloDisposable.java]: 100%. I wanted to do this actually and forgot to finish it 🤦

**Comment by martinbonnin** [comment]
(sorry, a bad rebase closed this PR automatically, I just force pushed the "good" rebase)

**Review by BoD — APPROVED** [review]

**Pull request merged**